presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -165,7 +166,12 @@ public Slice getSlice(int field)
             if (column.getType().getBase() == ColumnType.Base.DECIMAL) {
                 return (Slice) Decimals.parse(row.get(column.getPosition())).getObject();
             }
-            return Slices.utf8Slice(row.get(columns.get(field).getPosition()));
+
+            Slice characters = Slices.utf8Slice(row.get(columns.get(field).getPosition()));
+            if (column.getType().getBase() == ColumnType.Base.CHAR) {
```

Comment:
We don't check it for varchar, assuming that generator generates correct values ( that fit into the defined column size). The trick with trimming is quite different. Some values yield by generator end with a space. And due to the Presto internal representation of CHAR there should be no trailing spaces. 
For example

'lore ipsum ' - is a valid string for both, CHAR and VARCHAR. However when the representation of this value for VARCHAR is just the same, for CHAR trailing spaces are unacceptable.